### PR TITLE
Bar105 cpp 2a

### PR DESCRIPTION
**Pull request by lhog**

### CODE REVIEW

File: rts/Lua/LuaVAOImpl.cpp [hunk 1]
```diff
@@ -164,13 +163,12 @@ int LuaVAOImpl::AddObjectsToSubmissionImpl(const sol::stack_table& ids)
...
-		std::nullopt,	             // baseIndex
-		submitCmds.size() + idsSize, // instCount
-		std::nullopt,                // baseInstance
-	};
+	DrawCheckInput inputs = {};
+	inputs.drawCount    = std::nullopt;
+	inputs.baseVertex   = std::nullopt;
+	inputs.baseIndex    = std::nullopt;
+	inputs.instCount    = submitCmds.size() + idsSize;
+	inputs.baseInstance = std::nullopt;
```
Comment (sprunk) [inline rts/Lua/LuaVAOImpl.cpp:171]: perhaps there should be a constructor that takes these 5 args?

Comment (lhog) [reply rts/Lua/LuaVAOImpl.cpp:171]: Yes, I don't know why the previous code stopped working on 17->20 transition, I'm not proud with the fix, so any suggestions are welcome.

**Comment by sprunk** [comment]
btw DrawCheckInput's aggregate initialisation became invalid because the rules now prohibit `DrawCheckType() = default;`

**Comment by p2004a** [comment]
One issue, that pops up in clang in many places because it's in header file:

```
In file included from /home/p2004a/Workspace/BAR/spring/rts/lib/lua/include/LuaUser.cpp:10:
In file included from /home/p2004a/Workspace/BAR/spring/rts/Lua/LuaHandle.h:7:
In file included from /home/p2004a/Workspace/BAR/spring/rts/System/EventClient.h:12:
In file included from /home/p2004a/Workspace/BAR/spring/rts/System/Misc/SpringTime.h:12:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2320:48: error: call to consteval function 'std::chrono::hh_mm_ss::_S_fractional_width' is not a constant expression
        static constexpr unsigned fractional_width = {_S_fractional_width()};
                                                      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2320:48: note: undefined function '_S_fractional_width' cannot be used in a constant expression
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2275:2: note: declared here
        _S_fractional_width()
```

**Comment by p2004a** [comment]
Ok, ignore above, looks to be debian/clang/libstdc++ bug because this doesn't compile under clang with -std=c++20 either:

```cpp
#include <chrono>
int main() {return 0;}
```

**Comment by lhog** [comment]
I've reverted c++20 cmakelists instructions, but the rest of the changes should be safe to merge.

**Pull request merged**